plugins/vppv2/aclplugin/descriptor/acl.go

```diff
@@ -0,0 +1,397 @@
+				}
+			}
+		} else {
+			aclOldInInterfaces := c.getInterfaces(oldACL.Interfaces.Ingress)
+			aclOldEgInterfaces := c.getInterfaces(oldACL.Interfaces.Egress)
+			aclNewInInterfaces := c.getOrCacheInterfaces(newACL.Interfaces.Ingress, vppACLIndex, INGRESS)
+			aclNewEgInterfaces := c.getOrCacheInterfaces(newACL.Interfaces.Egress, vppACLIndex, EGRESS)
+			addedInInterfaces, removedInInterfaces := diffInterfaces(aclOldInInterfaces, aclNewInInterfaces)
+			addedEgInterfaces, removedEgInterfaces := diffInterfaces(aclOldEgInterfaces, aclNewEgInterfaces)
+
+			if len(removedInInterfaces) > 0 {
+				err := c.aclHandler.RemoveIPIngressACLFromInterfaces(vppACLIndex, removedInInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to remove IP ingress interfaces from ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+			if len(removedEgInterfaces) > 0 {
+				err := c.aclHandler.RemoveIPEgressACLFromInterfaces(vppACLIndex, removedEgInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to remove IP egress interfaces from ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+			if len(addedInInterfaces) > 0 {
+				err := c.aclHandler.SetACLToInterfacesAsIngress(vppACLIndex, addedInInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to set IP ingress interfaces to ACL %s: %v",
+						newACL.AclName, err)
+				}
+			}
+			if len(addedEgInterfaces) > 0 {
+				err := c.aclHandler.SetACLToInterfacesAsEgress(vppACLIndex, addedEgInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to add IP egress interfaces to ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+		}*/
+	}
+
+	return
+}
+
+func (d *AclDescriptor) ModifyWithRecreate(key string, oldACL, newACL *acl.Acl, metadata *aclidx.AclMetadata) bool {
+	var hasL2 bool
+	for _, rule := range oldACL.Rules {
+		if rule.GetMatch().GetMacipRule() != nil {
+			hasL2 = true
+		} else if rule.GetMatch().GetIpRule() != nil && hasL2 {
+			return true
+		}
+	}
+	return false
+}
+
+/*func (d *AclDescriptor) Update(key string, value *acl.Acl, metadata *aclidx.AclMetadata) error {
+
+}*/
+
+func (d *AclDescriptor) Dependencies(key string, value *acl.Acl) []api.Dependency {
+	return nil
+}
+
+func (d *AclDescriptor) DerivedValues(key string, value *acl.Acl) []api.KeyValuePair {
+	return nil
+}
+
+func (d *AclDescriptor) Dump(correlate []adapter.AclKVWithMetadata) (dump []adapter.AclKVWithMetadata, err error) {
+	ipACLs, err := d.aclHandler.DumpIPACL(d.ifPlugin.GetInterfaceIndex())
+	if err != nil {
+		return nil, errors.Errorf("failed to dump IP ACLs: %v", err)
+	}
+	macipACLs, err := d.aclHandler.DumpMACIPACL(d.ifPlugin.GetInterfaceIndex())
+	if err != nil {
+		return nil, errors.Errorf("failed to dump MAC IP ACLs: %v", err)
+	}
+	for _, ipACL := range ipACLs {
+		dump = append(dump, adapter.AclKVWithMetadata{
+			Key:   acl.Key(ipACL.ACL.Name),
+			Value: ipACL.ACL,
+			Metadata: &aclidx.AclMetadata{
+				Index: ipACL.Meta.Index,
+			},
+			Origin: api.FromSB,
```

Comment:
Origin: api.FromNB (configured by agent)

FromSB is for stuff added automatically that we do not want to touch (e.g.  automatically added routes, arps, loop0, ...)

Reply:
Although, how can we figure out if it was added from NB? Comparing it with correlate? What if it isn't there? UnknownOrigin?